src/App/Tracker/Controller/Hooks/Listeners/JoomlacmsPullsListener.php

```diff
@@ -93,99 +93,26 @@ public function onPullAfterUpdate(Event $event)
 	 */
-	protected function addRTClabel($hookData, Github $github, Logger $logger, $project, IssuesTable $table)
+	protected function checkRTClabel($hookData, Github $github, Logger $logger, $project, IssuesTable $table)
 	{
-		// Validation, if the status isn't RTC then go no further
-		if ($table->status != 4)
-		{
-			return;
-		}
-
 		// Set some data
-		$RTClabel    = 'RTC';
-		$addLabels   = array();
-		$rtcLabelSet = false;
+		$label      = 'RTC';
+		$labels     = array();
+		$labelIsSet = $this->checkLabel($hookData, Github $github, Logger $logger, $project, IssuesTable $table, $label);
 
-		// Get the labels for the pull's issue
-		try
+		// Validation, if the status isn't RTC or the Label is set then go no further
+		if ($labelIsSet == true && $table->status != 4)
 		{
-			$labels = $github->issues->get($project->gh_user, $project->gh_project, $hookData->pull_request->number)->labels;
+			// Remove the RTC label as it isn't longer set to RTC
+			$labels[] = $label;
+			$this->removeLabel($hookData, Github $github, Logger $logger, $project, IssuesTable $table, $labels);
 		}
-		catch (\DomainException $e)
-		{
-			$logger->error(
-				sprintf(
-					'Error retrieving labels for GitHub item %s/%s #%d - %s',
-					$project->gh_user,
-					$project->gh_project,
-					$hookData->pull_request->number,
-					$e->getMessage()
-				)
-			);
 
-			return;
-		}
-
-		// Check if the PR- label present if there are already labels attached to the item
-		if (count($labels) > 0)
+		if ($labelIsSet == false && $table->status == 4)
 		{
-			foreach ($labels as $label)
-			{
-				if (!$rtcLabelSet && $label->name == $RTClabel)
-				{
-					$logger->info(
-						sprintf(
-							'GitHub item %s/%s #%d already has the %s label.',
-							$project->gh_user,
-							$project->gh_project,
-							$hookData->pull_request->number,
-							$RTClabel
-						)
-					);
-
-					$rtcLabelSet = true;
-				}
-			}
-		}
-
-		// Add the RTC label if it isn't already set
-		if (!$rtcLabelSet)
-		{
-			$addLabels[] = $RTClabel;
-		}
-
-		// Only try to add labels if the array isn't empty
-		if (!empty($addLabels))
-		{
-			try
-			{
-				$github->issues->labels->add(
-					$project->gh_user, $project->gh_project, $hookData->pull_request->number, $addLabels
-				);
-
-				// Post the new label on the object
-				$logger->info(
-					sprintf(
-						'Added %s labels to %s/%s #%d',
-						count($addLabels),
-						$project->gh_user,
-						$project->gh_project,
-						$hookData->pull_request->number
-					)
-				);
-			}
-			catch (\DomainException $e)
-			{
-				$logger->error(
-					sprintf(
-						'Error adding labels to GitHub pull request %s/%s #%d - %s',
-						$project->gh_user,
-						$project->gh_project,
-						$hookData->pull_request->number,
-						$e->getMessage()
-					)
-				);
-			}
+			// Add the RTC label as it isn't already set
+			$labels[] = $label;
+			$this->addLabels($hookData, Github $github, Logger $logger, $project, IssuesTable $table, $labels);
```

Comment:
You should remove type hinting here. And also do not forget to remove $table as it is not used.